src/Uuid/ModelUuid.php

```diff
@@ -87,32 +116,24 @@ public function id(): string
 
 		// overwrite the content in memory and in the content file;
 		// use the most basic write method to avoid object cloning
-		$this->model->content()->update($data);
-		$this->model->writeContent($data);
+		$model->content('default')->update($data);
+		$model->writeContent($data, 'default');
+
+		// update the default translation
+		if ($kirby->multilang() === true) {
+			$model->translation('default')->update($data);
+		}
 
+		// switch back to the original user
 		$kirby->impersonate($user);
 
 		// TODO: replace the above in 3.9.0 with
 		// App::instance()->impersonate(
 		// 	'kirby',
-		// 	fn () => $this->model = $this->model->writeContent($data)
+		// 	fn () => $model->writeContent($data, $languageCode)
 		// );
 
-		// update the Uri object
-		$this->uri->host($id);
-
-		return $id;
-	}
-
-	/**
-	 * Retrieves the ID string (UUID without scheme) for the model
-	 * from the content file, if it is already stored there
-	 *
-	 * @param \Kirby\Cms\ModelWithContent $model
-	 */
-	public static function retrieveId(Identifiable $model): string|null
-	{
-		return $model->content()->get('uuid')->value();
+		return $model;
```

Comment:
Do we need to return the `$model`? Since it is no longer cloned during the update, I feel like we can get away with a `void` return type.

Reply:
I think something's wrong in the diff here. It's not returning just the model. It's returning the uuid vale.

Reply:
The `$model` is returned from `storeId()`, the `retrieveId()` method only appears in the diff because it was moved upwards.